source/common/http/header_map_impl.h

```diff
@@ -33,6 +33,18 @@ public:
     cached_byte_size_.reset();                                                                     \
     return maybeCreateInline(&inline_headers_.name##_, Headers::get().name);                       \
   }                                                                                                \
+  void setReference##name(const std::string& value) override {                                     \
+    addSize(value.size());                                                                         \
```

Comment:
Won't this double add if this is a new header per my above comment on the new addSize() call? If the header does exist, won't it not correctly account for the existing header size it is overwriting?

Reply:
Nope, because `maybeCreateInline(&inline_headers_.name##_, Headers::get().name)` will only account for the added key size. This call will add the value size.

Reply:
No! I see your point. Yes!!! Will fix, and add a testcase for setting reference and re-setting reference.